lib/app_profiler/backend/stackprof_backend.rb

```diff
@@ -35,31 +44,38 @@ def start(params = {})
         AppProfiler.logger.info(
           "[Profiler] failed to start the profiler error_class=#{error.class} error_message=#{error.message}"
         )
+        release_run_lock
         # This is a boolean instead of nil because StackProf#start returns a
         # boolean as well.
         false
       end
 
       def stop
         StackProf.stop
+      ensure
+        release_run_lock
       end
 
       def results
-        stackprof_profile = stackprof_results
+        stackprof_profile = backend_results
 
         return unless stackprof_profile
 
-        Profile.from_stackprof(stackprof_profile)
+        AppProfiler::AbstractProfile.from_stackprof(stackprof_profile)
```

Comment:
```suggestion
        AbstractProfile.from_stackprof(stackprof_profile)
```
🤔 I'm not a huge fan of calling methods from an abstract class. I wonder if we're better off keeping the name `Profile`, but we can fix this in another PR.

Reply:
@gmcgibbon this is the reason the deprecator was introduced, so if we go back to just "Profile" we don't need the deprecator stuff anymore.

A couple of ways to do this I can think of:

- Just create a subclass of "BaseProfile" called "Profile"
- Alias "BaseProfile" to "Profile"
- Make BaseProfile a module, include that in "Profile", and make "Profile" the base class
- Instead of calling it BaseProfile, just call it Profile

I don't really care or have a preference as to which, what do you think is the most idiomatic?

I agree that both "AbstractProfile.from..." and "BaseProfile.from..." look weird though.

Reply:
Let's rename to `BaseProfile` so that it is more aligned with how the rest of the gem is structured. You can call the from... methods in the subclasses or the base class, which I think it fine. I just don't like the "Abstract" naming here because it doesn't sound like Ruby to me.